android-json-form-wizard/src/test/java/com/vijay/jsonwizard/filesource/AssetsFileSourceTest.java

```diff
@@ -0,0 +1,88 @@
+package com.vijay.jsonwizard.filesource;
+
+import android.content.Context;
+import android.content.res.AssetManager;
+
+import org.jeasy.rules.api.Rules;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AssetsFileSourceTest {
+    @Mock
+    private Context context;
+
+    @Mock
+    private AssetManager assetManager;
+
+    private AssetsFileSource assetsFileSource = AssetsFileSource.INSTANCE;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void testInstanceIsNoNull() {
+        Assert.assertNotNull(AssetsFileSource.INSTANCE);
+    }
+
+    @Test
+    public void testGetRulesFromFileConvertsFileToRules() throws Exception {
+        String specifiedString = "1";
+        String relevance = "---\n" +
+                "name: step1_last_name\n" +
+                "description: last_name\n" +
+                "priority: 1\n" +
+                "condition: \"step1_first_Name.equalsIgnoreCase('Doe')\"\n" +
+                "actions:\n" +
+                "    - \"calculation = " + specifiedString + "\"";
+
+
+        Mockito.when(context.getAssets()).thenReturn(assetManager);
+        InputStream inputStream = new ByteArrayInputStream(relevance.getBytes());
+        Mockito.when(assetManager.open("rule/test")).thenReturn(inputStream);
+
+        Rules rules = assetsFileSource.getRulesFromFile(context, "rule/test");
+        Assert.assertFalse(rules.isEmpty());
```

Comment:
Wouldn't it be better to verify that the content is as expected and not just that it's not empty?

Reply:
The responsibility of the interface is to return the Rules Object when given a rule but not to verify that the Rule object works. i.e Given a syntactically correct input of type A, you expect an output or type B